packages/checkout-storefront/src/hooks/useFormSubmit/useFormSubmit.ts

```diff
@@ -0,0 +1,131 @@
+import { useAlerts } from "@/checkout-storefront/hooks/useAlerts";
+import { useCheckout } from "@/checkout-storefront/hooks/useCheckout";
+import { useLocale } from "@/checkout-storefront/hooks/useLocale";
+import {
+  CheckoutUpdateStateScope,
+  useCheckoutUpdateStateChange,
+} from "@/checkout-storefront/state/updateStateStore";
+import { useCallback } from "react";
+import { FormDataBase, FormHelpers } from "@/checkout-storefront/hooks/useForm";
+import {
+  CommonVars,
+  MutationBaseFn,
+  MutationData,
+  MutationVars,
+  ParserFunction,
+} from "@/checkout-storefront/hooks/useSubmit/types";
+import { ApiErrors } from "@/checkout-storefront/hooks/useGetParsedErrors/types";
+import { useGetParsedErrors } from "@/checkout-storefront/hooks/useGetParsedErrors";
+import { extractMutationErrors } from "@/checkout-storefront/lib/utils/common";
+import { localeToLanguageCode } from "@/checkout-storefront/lib/utils/locale";
+import { FormSubmitFn } from "@/checkout-storefront/hooks/useFormSubmit/types";
+
+interface CallbackProps<TData extends FormDataBase> {
+  formHelpers: FormHelpers<TData>;
+  formData: TData;
+}
+
+interface UseFormSubmitProps<TData extends FormDataBase, TMutationFn extends MutationBaseFn> {
+  scope: CheckoutUpdateStateScope;
+  onSubmit: (vars: MutationVars<TMutationFn>) => Promise<MutationData<TMutationFn>>;
+  parse: ParserFunction<TData, TMutationFn>;
+  onAbort?: (props: CallbackProps<TData>) => void;
+  onSuccess?: (props: CallbackProps<TData> & { result: MutationData<TMutationFn> }) => void;
+  onError?: (
+    props: CallbackProps<TData> & {
+      errors: ApiErrors<TData>;
+    }
+  ) => void;
+  onEnter?: (props: CallbackProps<TData>) => void;
```

Comment:
I got confused by this name a little bit - it's called when the submit starts, maybe we should call it `onSubmitStart` or `onStartSubmit`? I though it was on form enter (i.e. focus 🙈)

Reply:
Oh okay 👍 I guess in my mind it was more like "Enter the dragon" enter but how you perceived it makes perfect sense 😅 Will change